lib/browser/api/app.js

```diff
@@ -77,6 +79,28 @@ app.allowNTLMCredentialsForAllDomains = function (allow) {
   }
 }
 
+const makeSingleInstance = app.makeSingleInstance
+app.makeSingleInstance = function (callback) {
+  return makeSingleInstance.call(app, function (argv, ...args) {
+    if (process.platform === 'win32') {
+      // Delay the checking so the callback is run as quickly as we would expect
+      setImmediate(() => {
+        argv.forEach((arg) => {
+          const parsedURL = url.parse(arg)
+          // The URL needs a protocol and the "//" after the protocol
+          if (!parsedURL.protocol || !parsedURL.slashes) return
+          // The node URL module returns the protocol as "foo-bar:", we need to strip that colon
+          const protocol = parsedURL.protocol.substr(0, parsedURL.protocol.length - 1)
+          if (app.getDefaultProtocolClient(protocol).split(/ /g)[0].replace(/^"(.+)"$/g, '$1') === process.execPath) {
```

Comment:
Just wondering, does this work if a custom path was specified to http://electron.atom.io/docs/api/app/#appsetasdefaultprotocolclientprotocol-path-args-macos-windows ?